guava/src/com/google/common/collect/MoreCollectors.java

```diff
@@ -51,10 +51,12 @@
           Collector.Characteristics.UNORDERED);
 
   /**
-   * A collector that converts a stream of zero or one elements to an {@code Optional}. The returned
-   * collector throws an {@code IllegalArgumentException} if the stream consists of two or more
-   * elements, and a {@code NullPointerException} if the stream consists of exactly one element,
-   * which is null.
+   * A collector that converts a stream of zero or one elements to an {@code Optional}.
+   *
+   * @throws IllegalArgumentException if the stream consists of two or more elements.
+   * @throws NullPointerException if any element in stream is {@code null}.
```

Comment:
Same as https://github.com/google/guava/pull/5667#discussion_r682936112:
```suggestion
   * @throws NullPointerException if any element in the stream is {@code null}.
```